File: 0001_0100/0087/0087.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
	lines = [s[:-1] for s in f.readlines()[:-1]]
	lines.sort()
	m = 0
	for i in range(1, len(lines)):
		for j in range(i):
			if lines[i].startswith(lines[j]) and lines[i][len(lines[j]):] in lines:
				m += 1
				break
	q.write(str(m))

File: 0001_0100/0087/0087.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"sort"
	"strings"
)

func main() {
	c, _ := ioutil.ReadFile("input.txt")
	q, _ := os.Create("output.txt")
	lines := strings.Fields(string(c))
	n, m := len(lines)-1, 0
	lines = lines[:n]
	sort.Strings(lines)
	p := make(map[string]bool)
	for _, s := range lines {
		p[s] = true
	}
	for i := 1; i < n; i++ {
		for j := 0; j < i; j++ {
			if strings.HasPrefix(lines[i], lines[j]) && p[lines[i][len(lines[j]):]] {
				m++
				break
			}
		}
	}
	fmt.Fprint(q, m)
}
